### PR TITLE
Add support for destroy_data and update_data to be sent

### PR DESCRIPTION
**Pull request by jgrumboe**
This PR is inspired by #177 and it adds one feature:
- destroy_data => Allows specifying a 'data/body' to send along with destroy/delete requests.

It could solve #174 I think.

### CODE REVIEW

**Comment by jgrumboe** [comment]
@DRuggeri Is this PR ok? Is the test sufficient?

If it's fine I could also "extend" this PR to solve #124 and add `update_data` in the same way.

**Comment by m-garrido** [comment]
It would be awesome to have the same feature for update and delete.

**Comment by jgrumboe** [comment]
@m-garrido @melt-melt
I also added support for `update_data`.
@DRuggeri can you have a look at this?

**Comment by jgrumboe** [comment]
@DRuggeri I needed to update the GH release action also following this comment https://github.com/hashicorp/ghaction-import-gpg/issues/11#issuecomment-1185107410
Otherwise I wasn't able to publish the provider interim to my personal TF registry account.

I can remove the commit before merging.

**Comment by jgrumboe** [comment]
@m-garrido @melt-melt
I published my changes interim to TF registry under https://registry.terraform.io/providers/jgrumboe/restapi/1.18.0-dev
You could try out `update_data` and `destroy_data` for tests, would like to remove the provider from my account in the TF registry after this PR is merged.

**Review by arun-a-nayagam — COMMENTED** [review]
I have tested the update_data changes in an actual project and can confirm that it works fine.

**Comment by jgrumboe** [comment]
@DRuggeri can you help me here to get that merged? If something is missing like tests or docs, just tell me. Thanks in advance 🙏

**Comment by jgrumboe** [comment]
@douernesto @Martina-May 
Could you probably help in getting this PR merged? Unfortunately there's no feedback yet from @DRuggeri 😞

**Comment by Sh1ftry** [comment]
Hello, I am also waiting for this feature. Would be nice to have it released soon.

**Review by DRuggeri — CHANGES_REQUESTED** [review]
This looks good, though I am concerned about some of the changes to the release workflow. Can you please address those?

**Comment by DRuggeri** [comment]
Many apologies for the delay responding to this PR. This seems to make a lot of sense and I've added a quick few notes. If you can address those, we can go ahead and merge and cut a new release with this!

File: .github/workflows/release.yaml
Comment (jgrumboe) [reply .github/workflows/release.yaml:37]: This change is encouraged by one of hashicorp gh actions maintainer: https://github.com/hashicorp/ghaction-import-gpg/issues/11#issuecomment-1185107410
(also commented it here some time ago)

File: .github/workflows/release.yaml
Comment (jgrumboe) [reply .github/workflows/release.yaml]: Fixed in 225bd5f

File: .github/workflows/release.yaml
Comment (jgrumboe) [reply .github/workflows/release.yaml]: Fixed in 225bd5f

File: restapi/api_object.go
Comment (jgrumboe) [reply restapi/api_object.go]: Fixed in cb4ab82

File: docs/resources/object.md
Comment (jgrumboe) [reply docs/resources/object.md]: Fixed in cb4ab82

File: docs/resources/object.md
Comment (jgrumboe) [reply docs/resources/object.md]: Fixed in cb4ab82

**Comment by jgrumboe** [comment]
@DRuggeri Thanks for the feedback. I addressed it.

File: .github/workflows/release.yaml
Comment (DRuggeri) [reply .github/workflows/release.yaml:37]: Got it - thanks!

**Review by DRuggeri — APPROVED** [review]

**Pull request merged**
